samples/matter/common/src/event_triggers/event_triggers.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "event_triggers.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace Nrf::Matter;
+using namespace chip;
+
+CHIP_ERROR TestEventTrigger::RegisterTestEventTrigger(EventTrigger trigger)
+{
+	VerifyOrReturnError(trigger.Callback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(GetTriggerSlot(trigger.Id) == kInvalidTriggerSlot, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(mTriggersMap.FreeSlots() > 0, CHIP_ERROR_NO_MEMORY);
+
+	uint16_t freeSlot = mTriggersMap.GetFirstFreeSlot();
+	VerifyOrReturnError(freeSlot < kMaxEventTriggers, CHIP_ERROR_NO_MEMORY);
+
+	mTriggersMap.Insert(freeSlot, std::move(trigger));
+
+	LOG_INF("Registered new test event: 0x%llx on slot %d", trigger.Id, freeSlot);
+
+	return CHIP_NO_ERROR;
+}
+
+void TestEventTrigger::UnregisterTestEventTrigger(EventTriggerId id)
+{
+	TriggerSlot slot = GetTriggerSlot(id);
+	if (slot == kInvalidTriggerSlot) {
+		return;
+	}
+
+	LOG_INF("Unregistered test event: 0x%llx", id);
+
+	mTriggersMap.Erase(slot);
+}
+
+bool TestEventTrigger::DoesEnableKeyMatch(const chip::ByteSpan &enableKey) const
+{
+	return !mEnableKey.empty() && mEnableKey.data_equal(enableKey);
+}
+
+CHIP_ERROR TestEventTrigger::SetEnableKey(const chip::ByteSpan &newEnableKey)
+{
+	VerifyOrReturnError(newEnableKey.size() == chip::TestEventTriggerDelegate::kEnableKeyLength,
+			    CHIP_ERROR_INVALID_ARGUMENT);
+	return CopySpanToMutableSpan(newEnableKey, mEnableKey);
+}
+
+CHIP_ERROR TestEventTrigger::HandleEventTrigger(uint64_t eventTrigger)
+{
+	/* Check if the Enable Key is set to the "disabled" value. */
+	if (!IsEventTriggerEnabled()) {
+		return CHIP_ERROR_INCORRECT_STATE;
+	}
+
+	TriggerSlot slot = GetTriggerSlot(eventTrigger);
+
+	VerifyOrReturnError(slot != kInvalidTriggerSlot, CHIP_ERROR_INVALID_ARGUMENT);
+	VerifyOrReturnError(mTriggersMap[slot].Callback != nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Before using []operator from the FiniteMap, we must first check if the element is present in the map with Contains(), that's the current FiniteMap API limitation.